packages/mamba/PKGBUILD

```diff
@@ -22,7 +23,7 @@ build() {
 }
 
 package() {
-  depends+=('libasound.so' 'libfluidsynth.so' 'libjack.so' 'liblo.so' 'libsigc++' 'libsmf.so')
+  depends+=('libasound.so' 'libfluidsynth.so' 'libjack.so' 'liblo.so' 'libsigc++' 'libsmf.so' 'gcc-libs')
```

Comment:
AFAICS, @dvzrv only adds soname dependencies within the  `package` function and puts all other (like `gcc-libs`) in the `depends` in the file header.Maybe he can clarify what the strategy is here.

Reply:
The short answer is: It depends.

The longer answer is:
* Makepkg might detect the wrong package providing a given soname, if e.g. thinking in the context of lib32-foo in [multilib] providing `libfoo.so.1-32` and `foo` in [core], [extra] or [community] providing `libfoo.so.1-64`. 
  This is the case when only using a soname in the initial depends array. An alternative can be to first state the package, then the soname it provides (e.g. `depends=(foo libfoo.so)` in the context of the initial depends declaration) which might allow makepkg to resolve this as well, but may have quirks.
* When adding specific depends for a split package one might want to only have a subset of packages (which may or may not provide a soname) in their respective depends. Think of e.g. a split package `foo` and `foo-docs`, where `foo` depends on something, but `foo-docs` does not.

Reply:
What happend to the `depends+=` in the latest version of the `PKGBUILD`?

Reply:
Oh, I see, these are now optional dependencies.

I still get lots of `namcap` complaint, but they seem all to be bogus. `java-runtime` is already covered by `java-runtime-openjdk` and the other are optional or transient deps.

The "lacks FULL RELRO" could be fixed some ELF tools, I believe.

```
tuxguitar W: ELF file ('usr/lib/tuxguitar/libicedtea-sound.so') lacks FULL RELRO, check LDFLAGS.
tuxguitar W: ELF file ('usr/lib/tuxguitar/libtuxguitar-alsa-jni.so') lacks FULL RELRO, check LDFLAGS.
tuxguitar W: ELF file ('usr/lib/tuxguitar/libtuxguitar-fluidsynth-jni.so') lacks FULL RELRO, check LDFLAGS.
tuxguitar W: ELF file ('usr/lib/tuxguitar/libtuxguitar-jack-jni.so') lacks FULL RELRO, check LDFLAGS.
tuxguitar E: Dependency java-runtime detected and not included (found class files usr/share/java/tuxguitar/commons-compress.jar, usr/share/java/tuxguitar/gervill.jar, usr/share/java/tuxguitar/icedtea-sound.jar, usr/share/java/tuxguitar/itext-pdf.jar, usr/share/java/tuxguitar/itext-xmlworker.jar, usr/share/java/tuxguitar/swt.jar, usr/share/java/tuxguitar/tuxguitar-alsa.jar, usr/share/java/tuxguitar/tuxguitar-ascii.jar, usr/share/java/tuxguitar/tuxguitar-awt-graphics.jar, usr/share/java/tuxguitar/tuxguitar-browser-ftp.jar, usr/share/java/tuxguitar/tuxguitar-community.jar, usr/share/java/tuxguitar/tuxguitar-compat.jar, usr/share/java/tuxguitar/tuxguitar-converter.jar, usr/share/java/tuxguitar/tuxguitar-editor-utils.jar, usr/share/java/tuxguitar/tuxguitar-fluidsynth.jar, usr/share/java/tuxguitar/tuxguitar-gervill.jar, usr/share/java/tuxguitar/tuxguitar-gm-settings.jar, usr/share/java/tuxguitar/tuxguitar-gm-utils.jar, usr/share/java/tuxguitar/tuxguitar-gpx.jar, usr/share/java/tuxguitar/tuxguitar-gtp-ui.jar, usr/share/java/tuxguitar/tuxguitar-gtp.jar, usr/share/java/tuxguitar/tuxguitar-image.jar, usr/share/java/tuxguitar/tuxguitar-jack-ui.jar, usr/share/java/tuxguitar/tuxguitar-jack.jar, usr/share/java/tuxguitar/tuxguitar-jsa.jar, usr/share/java/tuxguitar/tuxguitar-lib.jar, usr/share/java/tuxguitar/tuxguitar-lilypond-ui.jar, usr/share/java/tuxguitar/tuxguitar-lilypond.jar, usr/share/java/tuxguitar/tuxguitar-midi-ui.jar, usr/share/java/tuxguitar/tuxguitar-midi.jar, usr/share/java/tuxguitar/tuxguitar-musicxml.jar, usr/share/java/tuxguitar/tuxguitar-pdf-ui.jar, usr/share/java/tuxguitar/tuxguitar-pdf.jar, usr/share/java/tuxguitar/tuxguitar-ptb.jar, usr/share/java/tuxguitar/tuxguitar-svg.jar, usr/share/java/tuxguitar/tuxguitar-synth-export.jar, usr/share/java/tuxguitar/tuxguitar-synth-gervill.jar, usr/share/java/tuxguitar/tuxguitar-synth.jar, usr/share/java/tuxguitar/tuxguitar-tef.jar, usr/share/java/tuxguitar/tuxguitar-tuner.jar, usr/share/java/tuxguitar/tuxguitar-ui-toolkit-swt.jar, usr/share/java/tuxguitar/tuxguitar-ui-toolkit.jar, usr/share/java/tuxguitar/tuxguitar.jar)
tuxguitar W: Dependency fluidsynth detected but optional (libraries ['usr/lib/libfluidsynth.so.3'] needed in files ['usr/lib/tuxguitar/libtuxguitar-fluidsynth-jni.so'])
tuxguitar W: Dependency jack2 detected but optional (libraries ['usr/lib/libjack.so.0'] needed in files ['usr/lib/tuxguitar/libtuxguitar-jack-jni.so'])
tuxguitar W: Dependency libpulse detected but optional (libraries ['usr/lib/libpulse.so.0'] needed in files ['usr/lib/tuxguitar/libicedtea-sound.so'])
tuxguitar W: Dependency included and not needed ('libxtst')
tuxguitar W: Dependency glibc included but already satisfied
tuxguitar W: Dependency included and not needed ('java-runtime-openjdk')
tuxguitar W: Dependency included and not needed ('gtk3')
``

Reply:
@SpotlightKid this comment was on the mamba-related changes which are already in master ;) I'll go over the dependencies again some time later, there's still room for optimization (e.g. gtk3 is not needed during build etc)

Reply:
Oops. ;)